diff --git a/include/lldb/Core/Module.h b/include/lldb/Core/Module.h
index 127ddae..d703bca 100644
--- a/include/lldb/Core/Module.h
+++ b/include/lldb/Core/Module.h
@@ -704,6 +704,8 @@ public:
     virtual SectionList *
     GetSectionList ();
 
+    size_t GetModuleSize();
+
     //------------------------------------------------------------------
     /// Notify the module that the file addresses for the Sections have
     /// been updated.
diff --git a/source/Commands/CommandObjectProcess.cpp b/source/Commands/CommandObjectProcess.cpp
index e587ead..cc9f1ca 100644
--- a/source/Commands/CommandObjectProcess.cpp
+++ b/source/Commands/CommandObjectProcess.cpp
@@ -1086,11 +1086,22 @@ protected:
             m_interpreter.GetDebugger().GetTargetList().SetSelectedTarget(target_sp.get());
         }
         
+        if (command.GetArgumentCount() == 0)
+        {
+            command.SetCommandString("connect://127.0.0.1:4321");
+        }
+
         if (command.GetArgumentCount() == 1)
         {
             const char *plugin_name = NULL;
-            if (!m_options.plugin_name.empty())
-                plugin_name = m_options.plugin_name.c_str();
+             if (!m_options.plugin_name.empty())
+            {
+                  plugin_name = m_options.plugin_name.c_str();
+            }
+            else
+            {
+                plugin_name = "gdb-remote";
+            }
 
             const char *remote_url = command.GetArgumentAtIndex(0);
             process = target_sp->CreateProcess (m_interpreter.GetDebugger().GetListener(), plugin_name, NULL).get();
diff --git a/source/Commands/CommandObjectTarget.cpp b/source/Commands/CommandObjectTarget.cpp
index 448da0e..e50c434 100644
--- a/source/Commands/CommandObjectTarget.cpp
+++ b/source/Commands/CommandObjectTarget.cpp
@@ -3380,8 +3380,9 @@ protected:
         bool dump_object_name = false;
         if (m_options.m_format_array.empty())
         {
-            m_options.m_format_array.push_back(std::make_pair('u', 0));
+            // m_options.m_format_array.push_back(std::make_pair('u', 0));
             m_options.m_format_array.push_back(std::make_pair('h', 0));
+            m_options.m_format_array.push_back(std::make_pair('l', 0));
             m_options.m_format_array.push_back(std::make_pair('f', 0));
             m_options.m_format_array.push_back(std::make_pair('S', 0));
         }
@@ -3418,6 +3419,12 @@ protected:
                     dump_object_name = true;
                     break;
 
+                case 'l':
+                    {
+                        strm.Printf("0x%X", module->GetModuleSize());
+                    }
+                    break;
+
                 case 'h':
                 case 'o':
                     // Image header address
@@ -3442,12 +3449,12 @@ protected:
                                         if (format_char == 'o')
                                         {
                                             // Show the offset of slide for the image
-                                            strm.Printf ("0x%*.*" PRIx64, addr_nibble_width, addr_nibble_width, header_load_addr - header_addr.GetFileAddress());
+                                            strm.Printf ("0x%*.*" PRIX64, addr_nibble_width, addr_nibble_width, header_load_addr - header_addr.GetFileAddress());
                                         }
                                         else
                                         {
                                             // Show the load address of the image
-                                            strm.Printf ("0x%*.*" PRIx64, addr_nibble_width, addr_nibble_width, header_load_addr);
+                                            strm.Printf ("0x%*.*" PRIX64, addr_nibble_width, addr_nibble_width, header_load_addr);
                                         }
                                     }
                                     break;
diff --git a/source/Core/Module.cpp b/source/Core/Module.cpp
index eb0359d..a68f9ea 100644
--- a/source/Core/Module.cpp
+++ b/source/Core/Module.cpp
@@ -1337,6 +1337,24 @@ Module::GetSectionList()
     return m_sections_ap.get();
 }
 
+size_t Module::GetModuleSize()
+{
+    SectionList* sections = GetSectionList();
+    size_t moduleSize = 0;
+
+    if (!sections)
+        return moduleSize;
+
+    for (size_t i = 0, n = sections->GetSize(); i != n; ++i)
+    {
+        SectionSP s = sections->GetSectionAtIndex(i);
+        if (s)
+            moduleSize += s->GetByteSize();
+    }
+
+    return moduleSize;
+}
+
 void
 Module::SectionFileAddressesChanged ()
 {
diff --git a/source/Initialization/SystemInitializerCommon.cpp b/source/Initialization/SystemInitializerCommon.cpp
index 51f32a2..a07d6ad 100644
--- a/source/Initialization/SystemInitializerCommon.cpp
+++ b/source/Initialization/SystemInitializerCommon.cpp
@@ -38,10 +38,12 @@
 
 #if defined(__APPLE__)
 #include "Plugins/DynamicLoader/Darwin-Kernel/DynamicLoaderDarwinKernel.h"
-#include "Plugins/ObjectFile/Mach-O/ObjectFileMachO.h"
 #include "Plugins/Platform/MacOSX/PlatformDarwinKernel.h"
 #endif
 
+#include "Plugins/ObjectFile/Mach-O/ObjectFileMachO.h"
+#include "Plugins/ObjectFile/Mach-O/ObjectFileMachO.cpp"
+
 #if defined(__linux__)
 #include "Plugins/Process/POSIX/ProcessPOSIXLog.h"
 #endif
@@ -131,8 +133,8 @@ SystemInitializerCommon::Initialize()
 #if defined(__APPLE__)
     DynamicLoaderDarwinKernel::Initialize();
     PlatformDarwinKernel::Initialize();
-    ObjectFileMachO::Initialize();
 #endif
+    ObjectFileMachO::Initialize();
 #if defined(__linux__)
     static ConstString g_linux_log_name("linux");
     ProcessPOSIXLog::Initialize(g_linux_log_name);
@@ -172,9 +174,9 @@ SystemInitializerCommon::Terminate()
 
 #if defined(__APPLE__)
     DynamicLoaderDarwinKernel::Terminate();
-    ObjectFileMachO::Terminate();
     PlatformDarwinKernel::Terminate();
 #endif
+    ObjectFileMachO::Terminate();
 
 #if defined(__WIN32__)
     ProcessWindowsLog::Terminate();
diff --git a/source/Plugins/Process/Utility/RegisterContextDarwin_arm.cpp b/source/Plugins/Process/Utility/RegisterContextDarwin_arm.cpp
index 4138a6a..d8c77bf 100644
--- a/source/Plugins/Process/Utility/RegisterContextDarwin_arm.cpp
+++ b/source/Plugins/Process/Utility/RegisterContextDarwin_arm.cpp
@@ -7,7 +7,7 @@
 //
 //===----------------------------------------------------------------------===//
 
-#if defined(__APPLE__)
+#if 1 //defined(__APPLE__)
 
 #include "RegisterContextDarwin_arm.h"
 
diff --git a/source/Plugins/Process/Utility/RegisterContextDarwin_arm64.cpp b/source/Plugins/Process/Utility/RegisterContextDarwin_arm64.cpp
index e08a873..1da3edd 100644
--- a/source/Plugins/Process/Utility/RegisterContextDarwin_arm64.cpp
+++ b/source/Plugins/Process/Utility/RegisterContextDarwin_arm64.cpp
@@ -7,7 +7,7 @@
 //
 //===----------------------------------------------------------------------===//
 
-#if defined(__APPLE__)
+#if 1 //defined(__APPLE__)
 
 #include "RegisterContextDarwin_arm64.h"
 
